nav2_toolkit/src/pose_saver_node.cpp

```diff
@@ -0,0 +1,296 @@
+
+  initial_pose_pub_ = this->create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>("/initialpose", 10);
+  last_sub_count_ = 0;
+  amcl_monitor_timer_ = this->create_wall_timer(
+      2s,  // check every 2 seconds
+      std::bind(&PoseSaverNode::amcl_monitor_callback, this));
+
+  start_service_ = this->create_service<std_srvs::srv::Trigger>(
+    "/start_pose_saver",
+    std::bind(&PoseSaverNode::start_service_cb, this, std::placeholders::_1, std::placeholders::_2));
+
+  stop_service_ = this->create_service<std_srvs::srv::Trigger>(
+    "/stop_pose_saver",
+    std::bind(&PoseSaverNode::stop_service_cb, this, std::placeholders::_1, std::placeholders::_2));
+
+  restore_service_ = this->create_service<std_srvs::srv::Trigger>(
+    "/localise_at_last_known_position",
+    std::bind(&PoseSaverNode::restore_service_cb, this, std::placeholders::_1, std::placeholders::_2));
+
+  reset_service_ = this->create_service<std_srvs::srv::Trigger>(
+    "/reset_last_known_pose",
+    std::bind(&PoseSaverNode::reset_pose_file_cb, this, std::placeholders::_1, std::placeholders::_2));
+
+  if (auto_start) {
+    saving_active_ = true;
+    timer_->reset();
+    RCLCPP_INFO(this->get_logger(), "Pose saving auto-started.");
+  }
+
+  if (auto_restore_) {
+    RCLCPP_INFO(this->get_logger(), "Auto-restore enabled. Waiting for AMCL...");
+  
+    // Wait for sim time to become active
+    if (this->get_parameter_or("use_sim_time", false)) {
+      RCLCPP_INFO(this->get_logger(), "Waiting for /clock to start (sim time)...");
+      while (rclcpp::ok() && this->now().nanoseconds() == 0) {
+        rclcpp::sleep_for(100ms);
+      }
+    }
+  
+    // Wait for AMCL to subscribe to /initialpose
+    rclcpp::Time start = this->now();
+    rclcpp::Duration timeout = rclcpp::Duration::from_seconds(15.0);
+    while (initial_pose_pub_->get_subscription_count() == 0 &&
+           (this->now() - start) < timeout)
+    {
+      RCLCPP_WARN_THROTTLE(this->get_logger(), *this->get_clock(), 2000,
+                           "Waiting for AMCL");
+      rclcpp::sleep_for(200ms);
+    }
+  
+    if (initial_pose_pub_->get_subscription_count() == 0) {
+      RCLCPP_ERROR(this->get_logger(), "Timeout: AMCL did not start.");
+    } else {
+      try {
+        auto pose_msg = read_pose_from_file(pose_file_path_);
+        rclcpp::sleep_for(500ms);
+        initial_pose_pub_->publish(pose_msg);
+        RCLCPP_INFO(this->get_logger(), "Auto-restored initial pose from file.");
+      } catch (const std::exception &e) {
+        RCLCPP_WARN(this->get_logger(), "Failed auto-restore: %s", e.what());
+      }
+    }
+  }
+  
+
+  RCLCPP_INFO(this->get_logger(), "Pose Saver Node Initialized.");
+}
+
+void PoseSaverNode::pose_callback(const geometry_msgs::msg::PoseWithCovarianceStamped::SharedPtr msg)
+{
+  last_pose_ = msg;
+}
+
+void PoseSaverNode::timer_callback()
+{
+  if (!saving_active_ || !last_pose_) return;
+
+  try {
+    write_pose_to_file("/tmp/pose_saver.yaml");
```

Comment:
Save only in the set filepath param location, not a hardcoded location and not in `get_package_config_path`